storage/innobase/include/row0mysql.h

```diff
@@ -681,6 +681,12 @@ struct mysql_row_templ_t {
 					Innobase record in the current index;
 					not defined if template_type is
 					ROW_MYSQL_WHOLE_ROW */
+	bool	rec_field_is_prefix;	/* is this field in a prefix index? */
```

Comment:
OK, having prototyped it I think it's better to keep this flag. Making the change in dict_index_get_nth_col_or_prefix_pos to return prefix_col_pos == ULINT_UNDEFINED whenever pos != ULINT_UNDEFINED makes sure that build_template_field starts out with rec_field_is_prefix == (rec_field_no == ULINT_UNDEFINED && rec_prefix_field_no != ULINT_UNDEFINED) holding. However this stops holding at the end of ha_innobase::build_template where templ->rec_field_no is reset to templ->clust_rec_field_no, optionally, and it complicates the control flow everywhere.

The diff of the prototype is at https://gist.github.com/laurynas-biveinis/d57b618a909001d4eb575c9cea63e20f